### PR TITLE
[FLOC-1181] Unattached manifestations design sketch

### PR DESCRIPTION
**Pull request by itamarst**
https://clusterhq.atlassian.net/browse/FLOC-1181
Wooo :ghost:

Depends on FLOC-1159... or at least, I thought it did? Anyway it's a diff against that, fine for reviewing.

### CODE REVIEW

File: flocker/control/_model.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 """
 
 from characteristic import attributes, Attribute
-from pyrsistent import pmap
+from pyrsistent import pmap, pset
```
Comment (exarkun) [inline flocker/control/_model.py]: is pset better than frozenset?

Comment (itamarst) [inline flocker/control/_model.py]: It has an `add()` method? But who knows maybe not.

File: flocker/node/_deploy.py [hunk 2]
```diff
@@ -420,8 +420,8 @@ def applications_from_units(result):
                 running=running,
                 not_running=not_running,
                 used_ports=self.network.enumerate_used_ports(),
-                # FLOC-1181 will want to add here the manifestations that
-                # are not attached to any application.
+                # FLOC-1181 will want to add here the remaining
+                # manifestations that are not attached to any application.
```
Comment (exarkun) [inline flocker/node/_deploy.py]: That is to say, all the manifestations that the above loop didn't consume from `available_manifestations`?

File: flocker/control/_model.py (hunk 1 above)
Comment (exarkun) [inline flocker/control/_model.py]: That's probably useful eventually.  I'm happy to see pset used but I guess I'll be sad if we randomly use frozenset in some places and pset in others.  Maybe just stick with frozenset until pset is actually a win for something?

**Comment by exarkun** [comment]
Thanks, looks good, please proceed.

**Pull request closed without merging**